### PR TITLE
jv - add course occupancy by department statistics page

### LINKED ISSUE #44: Overall course occupancy (total_enrolled/total_capacity) in department
# User Story

As a user, I can select a start quarter, an end quarter, and a department to view the total regular course occupancy by the selected department. 

# Acceptance Criteria

- [x] The "Overall Course Occupancy" menu button directs to the total occupancy statistics page
- [x] On the page, there is a form with the user being able to select a start quarter, end quarter, and department
- [x] After submitting the form, the statistics are displayed at the bottom of the form

# Implementation Todos

Front end:

- [x] Add a "Overall Course Occupancy" page
- [x] Add a form with a drop down selection of start and end quarter and department
- [x] Add a results component that appears in the bottom with the resulting statistic

Back end:

- [x] Endpoint to retrieve occupancy information given a department, start and end quarters

Testing:

- [x] Front end tests pass and there is adequate coverage
- [x] Back end tests pass and there is adequate coverage

### PR DESCRIPTION
**Pull request by jvoucsb**
This PR implements the "Course Occupancy By Department" statistic page, where a user can select a start and end quarter along with the department to get the amount of enrolled, amount of max enrollment, and occupancy percentage across all regular classes in the department. Closes #44 

course-search-course-occupancy

### CODE REVIEW

File: javascript/src/main/components/Statistics/CourseOccupancyForm.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+const CourseOccupancyForm = ({ setOccupancyJson, fetchJSON, onSubmit = () => {} }) => {
+
+    const [startQuarter, setStartQuarter] = useState("20204");
+    const [endQuarter, setEndQuarter] = useState("20211");
+    const [department, setDepartment] = useState("CMPSC");
+    const [loading, setLoading] = useState(false);
+
+    const handleSubmit = (event) => {
+        event.preventDefault();
+        console.log("submit pressed");
```
Comment (zminshall) [inline javascript/src/main/components/Statistics/CourseOccupancyForm.js:15]: Should we keep this console.log?

**Review by zminshall — APPROVED** [review]
Looks good to me and I tested locally and found no bugs.

**Review by paritoshjha24 — DISMISSED** [review]
Deployed locally with no issues.  Code follows general coding practices well. Merge Away!

**Review by Byngyn — DISMISSED** [review]
Ran tests locally and all passed. No bugs found.

File: javascript/src/main/components/Statistics/CourseOccupancyForm.js (hunk 1 above)
Comment (pconrad) [reply javascript/src/main/components/Statistics/CourseOccupancyForm.js]: I think it can go bye bye

**Review by pconrad — APPROVED** [review]
LGTM

**Comment by pconrad** [comment]
40 points!

**Pull request merged**